src/lib.rs

```diff
@@ -29,3 +35,125 @@ mod violation;
 pub mod external {
     pub use ion_rs;
 }
+
+/// Provide an Ion schema element which includes all OwnedElements and a document type
+///
+/// ## Example:
+/// In general `TypeRef` `validate()` takes in IonSchemaElement as the value to be validated.
+/// In order to create an `IonSchemaElement`:
+///
+/// ```
+/// use ion_rs::value::owned::OwnedElement;
+/// use ion_schema::IonSchemaElement;
+///
+/// // create an IonSchemaElement from an OwnedElement
+/// let owned_element: OwnedElement = 4.into();
+/// let ion_schema_element: IonSchemaElement = (&owned_element).into();
+///
+/// // create an IonSchemaElement for document type based on vector of owned elements
+/// let document: IonSchemaElement = IonSchemaElement::Document(vec![owned_element]);
+/// ```
+#[derive(Debug, Clone)]
+pub enum IonSchemaElement {
+    SingleElement(OwnedElement),
+    Document(Vec<OwnedElement>),
+}
+
+impl IonSchemaElement {
+    pub fn as_element(&self) -> Option<&OwnedElement> {
+        match self {
+            IonSchemaElement::SingleElement(element) => Some(element),
+            IonSchemaElement::Document(_) => None,
+        }
+    }
+
+    pub fn as_document(&self) -> Option<&Vec<OwnedElement>> {
+        match self {
+            IonSchemaElement::SingleElement(_) => None,
+            IonSchemaElement::Document(document) => Some(document),
+        }
+    }
```

Comment:
question—is there any particular reason for creating these functions rather than using `From`/`Into`?